### ISSUE TITLE
How to preprocess Human dendritic cells dataset and set the hyperparameters for training

### ISSUE THREAD

**Issue by XiHuYan** [ml4bio/CLEAR#1]
Thanks for sharing the work!

I have trouble in reproducing the figure Fig.3a in your paper. Here are my reproduction steps,

Preprocessing human dendritic dataset with 4 pipelines:
   (a). directly select hvg, and then normalize each cell to 10000
   (b). the same as (a), then scale 
   (c). log transformed, normalize each cell to 10000, then select hvg 
   (d). the same as (c), then scale

training model with parameters:
     --epochs 100 --lr 0.01 --batch_size 128 --pcl_r 1024 --cos --gpu 0

after training, I take the embedded features to finish visualization(pca transformed, then umap)

the visualization from the first preprocessing pipeline is close to Fig.3a, but it's unsatisfactory.
![d1](https://user-images.githubusercontent.com/41266664/140938225-2d6c83dc-598c-4582-8d86-6a40a212ae67.PNG)

The pictures from the other 3 preprocessing pipelines are all the same.
![d1_2](https://user-images.githubusercontent.com/41266664/140938872-862a316c-5da3-4cd6-9e5b-7dcfce6a14c4.PNG)

So, how can I modify the preprocessing pipeline or set proper training parameters to reproduce the Fig.3a in your paper?

**Comment by mydkzgj** [comment]
Hi @XiHuYan  Thank you for your attention! You can try the following configuration for the reproduction of Fig.3a. 
Preprocess: --filter --norm --scale
Train: --epochs 200 --lr 1 --batch_size 512 --pcl_r 1024 --cos --eval_freq 1
Hope these can help you.

**Comment by XiHuYan** [comment]
Thanks for your reply!

I have tried your proposed configuration, and this is what I plotted use the feature embeddings,
![CLEAR_dendritic](https://user-images.githubusercontent.com/41266664/141397916-2f1ae2db-daf3-4d4b-ae1e-e1928bc3efa3.PNG)

It looks like getting close to Fig.3a. But I'm confused about the training parameters. There is only 576 cells in the human dendritics cell datasets(downloaded from the https://github.com/JinmiaoChenLab/Batch-effect-removal-benchmarking/tree/master/Data/dataset1), but the batch size is set 512, which means the model is trained nearly once in each iteration. Is that appropriate?

And, it seems that CLEAR needs distinct preprocessing steps for different datasets. Does it means that if I want to use CLEAR to complete batch effect correction I have to try all the preprocessing combinations to get the desired choice? Could you offer some suggestions for this?

thanks a lot!

**Comment by WHan-alter** [comment]
Dear @XiHuYan ,

Thank you for your reply! For the first issue, I think the batch size does not affect the model performance. Some recent studies about contrastive learning suggest that large batch sizes help contrastive models do better. For the second question, the problem is about quality control. It is a general problem in the field since we do not have a golden standard for it. (Although there are some popular protocols.(See ScanPy and Seurat preprocessing part) The settings you used is one of the protocol)

Best,
Wenkai
